### PR TITLE
[PM-3756] Disable node integration and enable context isolation in desktop

### PR DESCRIPTION
**Pull request by dani-garcia**
## Type of change
```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Disable node integration and enable context isolation for the renderer process. This requires removing the preload polyfill, enabling the context bridge, and changing the webpack target to avoid bringing in electron-only modules.

The electron log service had to be split into two because the renderer process cannot use utils.isDev(), as it requires access to the `process` module.

### CODE REVIEW

File: apps/desktop/src/platform/services/electron-platform-utils.service.ts [hunk 1]
```diff
@@ -94,7 +94,7 @@ export class ElectronPlatformUtilsService implements PlatformUtilsService {
   }
 
   isDev(): boolean {
-    return ipc.platform.isDev;
+    return isDev();
```
Comment (Hinton) [inline apps/desktop/src/platform/services/electron-platform-utils.service.ts:97]: Shouldn't this be using the ipc, same for isMacAppStore?

File: apps/desktop/webpack.renderer.js [hunk 2]
```diff
@@ -65,8 +69,7 @@ const renderer = {
   mode: NODE_ENV,
   devtool: "source-map",
   // TODO: Replace this with web.
```
Comment (Hinton) [inline apps/desktop/webpack.renderer.js:71]:
```suggestion
```

File: apps/desktop/src/platform/services/electron-log.renderer.service.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import { LogLevelType } from "@bitwarden/common/enums";
+import { ConsoleLogService as BaseLogService } from "@bitwarden/common/platform/services/console-log.service";
+
+export class ElectronLogRendererService extends BaseLogService {
+  constructor(protected filter: (level: LogLevelType) => boolean = null) {
+    super(ipc.platform.isDev, filter);
+  }
+
+  write(level: LogLevelType, message: string) {
```
Comment (Hinton) [inline apps/desktop/src/platform/services/electron-log.renderer.service.ts:11]: At this point would it make sense to just cut out the `electron-log` dependency in the renderer and use regular ipc messaging we own?

File: apps/desktop/src/platform/services/electron-platform-utils.service.ts (hunk 1 above)
Comment (dani-garcia) [reply apps/desktop/src/platform/services/electron-platform-utils.service.ts]: Yeah, I think the rebase somehow reset the isDev change. Also updated the isMacAppStore check

File: apps/desktop/src/app/accounts/settings.component.ts
Comment (Hinton) [inline apps/desktop/src/app/accounts/settings.component.ts:1]: These aren't exactly equivalent, please ensure QA tests this.

**Review by Hinton — APPROVED** [review]

File: apps/desktop/src/app/accounts/settings.component.ts
Comment (dani-garcia) [reply apps/desktop/src/app/accounts/settings.component.ts:1]: They should be the same, I think.`deviceType` is calculated in the preload like this: 
https://github.com/bitwarden/clients/blob/ed4cb06433b4371fe68373ef38fd73476eb42251/apps/desktop/src/platform/preload.ts#L126-L135

`ipc.platform.deviceType === DeviceType.LinuxDesktop` could be matching other platforms like `openbsd`, but those aren't supported.

**Review by Hinton — DISMISSED** [review]

File: apps/desktop/src/utils.ts [hunk 4]
```diff
@@ -53,7 +53,7 @@ export function isWindowsStore() {
   if (
     windows &&
     !windowsStore &&
-    process.resourcesPath.indexOf("8bitSolutionsLLC.bitwardendesktop_") > -1
+    process.resourcesPath?.indexOf("8bitSolutionsLLC.bitwardendesktop_") > -1
```
Comment (Hinton) [inline apps/desktop/src/utils.ts:56]: You could potentially call ipc to invoke this on main. I'm insure if it's still needed though.

**Review by Hinton — DISMISSED** [review]

File: apps/desktop/src/utils.ts (hunk 4 above)
Comment (dani-garcia) [reply apps/desktop/src/utils.ts:56]: I got the Windows Store appx built and running, and in that case process.windowsStore was set to true correctly, so maybe the workaround isn't needed yeah.

**Comment by dani-garcia** [comment]
The `fromBufferToUtf8` function had a separate node-specific branch, which wasn't entirely identical to the non-node variant.

The difference between them was due to how they handled non-utf8 encoded strings, the node variant had no troubles with that, while the non-node variant would fail when passing it through `decodeURIComponent`.

When enabling electron sandboxing, this has caused a code path that previously went through the node variant to fail, specifically here: https://github.com/bitwarden/clients/blob/25711afaf69c7a97ad78e5a333043f69b1f233f7/apps/desktop/src/platform/services/electron-crypto.service.ts#L123

I've switched the implementation to use the `buffer` package and so we avoid the conditional differences. I've also added some tests to ensure that the results match the expected values. I think it would be a good idea to change all the `Buffer` related functions in Utils to use this packages instead of having `isNode` conditions all over the place, but I'll leave that for later.

**Review by Hinton — DISMISSED** [review]

**Review by Hinton — APPROVED** [review]

**Pull request merged**